vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -210,16 +224,17 @@ def submit_metrics_callback(self, query_results):
                         to_string(metric_name),
                     )
                     continue
-                value = valid_values[-1]
-                if metric_name in PERCENT_METRICS:
-                    # Convert the percentage to a float.
-                    value /= 100.0
 
                 tags = []
-                if should_collect_per_instance_values(metric_name, resource_type):
-                    instance_tag_key = get_mapped_instance_tag(metric_name)
-                    instance_tag_value = result.id.instance or 'none'
-                    tags.append('{}:{}'.format(instance_tag_key, instance_tag_value))
+                if should_collect_per_instance_values(self.config, metric_name, resource_type):
```

Comment:
```suggestion
                if should_collect_per_instance_values(self.config, metric_name, resource_type) and metric_name in have_instance_value[resource_type]:
```

Reply:
I think it makes more sense to have it here. Or maybe add an extra kwargs to `should_collect_per_instance_values` ?

Reply:
Yes, indeed, can be simplified a bit. Done :)